src/make_trade.ts

```diff
@@ -9,6 +9,7 @@ import { TokenInfo, TokenList } from "@uniswap/token-lists";
 import { BigNumber, ethers } from "ethers";
 import { HardhatRuntimeEnvironment } from "hardhat/types";
 import fetch from "node-fetch";
+import shuffle from "shuffle-array";
```

Comment:
ubernit: This feels like one of those `leftPad` dependencies :stuck_out_tongue:. You could have used Knuth shuffle which is fairly simple:
```ts
function shuffle(array: Array): Array {
  for (let i = array.length - 1; i > 0; i--) {
    const j = ~~(Math.random() * (i + 1));
    ([array[i], array[j]] = [array[j], array[i]]);
  }
  return array;
}
```

Reply:
This is an interesting take. I always feel like when I'm implementing an interview question for real work, I'm doing something wrong (someone else has probably written it better than me and it will be less code to maintain) - so I'm trying to avoid these things. 
I assume your worry (and reference to leftPad) is that external dependencies are a risk which we should try to minimise?

Reply:
No, it just feels weird to pull in a dependency for this (hence the ubernit). It was fine as was though.